Extension and old Maven version

When we use njord as extension and execute project on old Maven version we have:

```
Maven home: /home/runner/.m2/wrapper/dists/apache-maven-3.6.3/5542c391
Java version: 1.8.0_452, vendor: Azul Systems, Inc., runtime: /opt/hostedtoolcache/Java_Zulu_jdk/8.0.452-[9](https://github.com/mojohaus/exec-maven-plugin/actions/runs/15224781972/job/42825623880#step:13:10)/x64/jre
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "6.11.0-1014-azure", arch: "amd64", family: "unix"
[INFO] Error stacktraces are turned on.
[INFO] Scanning for projects...
Error:  Error enabling Njord: Unable to provision, see the following errors:
Error:  
Error:  1) null returned by binding at org.eclipse.sisu.wire.LocatorWiring
Error:   but the 1st parameter of eu.maveniverse.maven.njord.shared.impl.DefaultSessionFactory.<init>(Unknown Source) is not @Nullable
Error:    at org.eclipse.sisu.wire.LocatorWiring
Error:    while locating eu.maveniverse.maven.njord.shared.impl.InternalArtifactStoreManagerFactory
Error:      for the 1st parameter of eu.maveniverse.maven.njord.shared.impl.DefaultSessionFactory.<init>(Unknown Source)
Error:    at ClassRealm[extension>eu.maveniverse.maven.njord:extension:0.6.0, parent: sun.misc.Launcher$AppClassLoader@7852e922] (via modules: org.eclipse.sisu.wire.WireModule -> org.eclipse.sisu.plexus.PlexusBindingModule)
Error:    while locating eu.maveniverse.maven.njord.shared.impl.DefaultSessionFactory
Error:    while locating java.lang.Object annotated with *
Error:    at org.eclipse.sisu.wire.LocatorWiring
Error:    while locating eu.maveniverse.maven.njord.shared.SessionFactory
Error: [ERROR] 
Error: [ERROR] 1 error
Error:  -> [Help 1]
org.apache.maven.MavenExecutionException: Error enabling Njord
    at eu.maveniverse.maven.njord.extension3.NjordSessionLifecycleParticipant.afterProjectsRead (NjordSessionLifecycleParticipant.java:60)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:264)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:[10](https://github.com/mojohaus/exec-maven-plugin/actions/runs/15224781972/job/42825623880#step:13:11)5)
```

It can be difficult for some users to guess what is a problem.

Extension is load at the beginning so project can not check, eg by enforcer plugin.

Can we detect such in extension, and generate proper message ... or even more make possibility to skip executing on old Maven ...

--- a/core/pom.xml
+++ b/core/pom.xml
@@ -30,42 +30,51 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
+      <scope>provided</scope>
     </dependency>
 
     <!-- Resolver -->
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-spi</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-util</artifactId>
+      <scope>provided</scope>
     </dependency>
 
     <!-- Maven -->
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-model</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-resolver-provider</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-core</artifactId>
+      <scope>provided</scope>
     </dependency>
 
     <dependency>

--- a/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
+++ b/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
@@ -58,7 +58,7 @@ public class NjordSessionLifecycleParticipant extends AbstractMavenLifecyclePart
             }
         } catch (Exception e) {
             if ("com.google.inject.ProvisionException".equals(e.getClass().getName())) {
-                logger.warn("Njord session creation failed"); // here runtime req will kick in
+                logger.error("Njord session creation failed", e); // here runtime req will kick in
             } else {
                 throw new MavenExecutionException("Error enabling Njord", e);
             }

--- a/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
+++ b/extension/src/main/java/eu/maveniverse/maven/njord/extension3/NjordSessionLifecycleParticipant.java
@@ -57,7 +57,11 @@ public class NjordSessionLifecycleParticipant extends AbstractMavenLifecyclePart
                 logger.info("Njord {} session created", ns.config().version().orElse("UNKNOWN"));
             }
         } catch (Exception e) {
-            throw new MavenExecutionException("Error enabling Njord", e);
+            if ("com.google.inject.ProvisionException".equals(e.getClass().getName())) {
+                logger.warn("Njord session creation failed"); // here runtime req will kick in
+            } else {
+                throw new MavenExecutionException("Error enabling Njord", e);
+            }
         }
     }
 

--- a/publisher/apache/pom.xml
+++ b/publisher/apache/pom.xml
@@ -25,6 +25,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>eu.maveniverse.maven.njord</groupId>
@@ -39,18 +40,22 @@
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-spi</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-util</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
     </dependency>
   </dependencies>
 </project>

--- a/publisher/deploy/pom.xml
+++ b/publisher/deploy/pom.xml
@@ -25,6 +25,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>eu.maveniverse.maven.njord</groupId>
@@ -34,19 +35,22 @@
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-spi</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-util</artifactId>
+      <scope>provided</scope>
     </dependency>
-
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
     </dependency>
   </dependencies>
 </project>

--- a/publisher/sonatype/pom.xml
+++ b/publisher/sonatype/pom.xml
@@ -25,6 +25,7 @@
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>eu.maveniverse.maven.njord</groupId>
@@ -34,14 +35,17 @@
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-api</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-spi</artifactId>
+      <scope>provided</scope>
     </dependency>
     <dependency>
       <groupId>org.apache.maven.resolver</groupId>
       <artifactId>maven-resolver-util</artifactId>
+      <scope>provided</scope>
     </dependency>
 
     <dependency>
@@ -74,59 +78,10 @@
       <groupId>org.slf4j</groupId>
       <artifactId>jcl-over-slf4j</artifactId>
     </dependency>
-
     <dependency>
       <groupId>javax.inject</groupId>
       <artifactId>javax.inject</artifactId>
+      <scope>provided</scope>
     </dependency>
   </dependencies>
-
-  <build>
-    <plugins>
-      <plugin>
-        <groupId>org.apache.maven.plugins</groupId>
-        <artifactId>maven-shade-plugin</artifactId>
-        <executions>
-          <execution>
-            <id>shade</id>
-            <goals>
-              <goal>shade</goal>
-            </goals>
-            <phase>package</phase>
-            <configuration>
-              <dependencyReducedPomLocation>${build.directory}/dependency-reduced-pom.xml</dependencyReducedPomLocation>
-              <artifactSet>
-                <includes>
-                  <include>org.apache.httpcomponents:*</include>
-                  <inclide>commons-codec:commons-codec</inclide>
-                  <include>org.slf4j:jcl-over-slf4j</include>
-                </includes>
-              </artifactSet>
-              <filters>
-                <filter>
-                  <artifact>*:*</artifact>
-                  <excludes>
-                    <exclude>META-INF/DEPENDENCIES</exclude>
-                    <exclude>META-INF/LICENSE</exclude>
-                    <exclude>META-INF/NOTICE</exclude>
-                    <exclude>META-INF/MANIFEST.MF</exclude>
-                  </excludes>
-                </filter>
-              </filters>
-              <relocations>
-                <relocation>
-                  <pattern>org.apache.http</pattern>
-                  <shadedPattern>eu.maveniverse.maven.njord.publisher.sonatype.shaded</shadedPattern>
-                </relocation>
-                <relocation>
-                  <pattern>org.apache.commons</pattern>
-                  <shadedPattern>eu.maveniverse.maven.njord.publisher.sonatype.shaded</shadedPattern>
-                </relocation>
-              </relocations>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-    </plugins>
-  </build>
 </project>

